src/test/java/io/cdap/delta/bigquery/BigQueryTargetTest.java

```diff
@@ -41,4 +113,122 @@ public void testStagingBucketName() {
     Assert.assertEquals(expectedBucketName, BigQueryTarget.getStagingBucketName("   ", pipelineId));
     Assert.assertEquals(expectedBucketName, BigQueryTarget.getStagingBucketName(null, pipelineId));
   }
+
+  @Test
+  public void testGetMaximumExistingSequenceNumberForRetryableFailures() throws Exception {
+    List<Throwable> exceptions = new ArrayList<>();
+    exceptions.add(new BigQueryException(500, null));
+    exceptions.add(new BigQueryException(BILLING_TIER_LIMIT_EXCEEDED_CODE, null,
+            new BigQueryError(BILLING_TIER_LIMIT_EXCEEDED_REASON, null, null)));
+    exceptions.add(new BigQueryException(RATE_LIMIT_EXCEEDED_CODES.stream().findAny().get(), null,
+            new BigQueryError(RATE_LIMIT_EXCEEDED_REASON, null, null)));
+
+    BigQueryTarget bqTarget = new BigQueryTarget(conf);
+
+    for (Throwable exception: exceptions) {
+      PowerMockito.mockStatic(BigQueryUtils.class);
+      PowerMockito.doThrow(exception).when(BigQueryUtils.class);
+      BigQueryUtils.getMaximumExistingSequenceNumber(Mockito.anySet(), Mockito.anyString(),
```

Comment:
To simulate real behaviour (RuntimeException thrown from the method), it would be better to not mock `BigQueryUtils.getMaximumExistingSequenceNumber` itself, but instead the internal BigQuery call, please refer `BigQueryConsumerTest.testPermanentFailureIsNotRetriedInProcessDDL` in latest code.